### PR TITLE
sub-menu component, --shadow-black custom property, icons.

### PR DESCRIPTION
**Pull request by Pete-Robelou**
- sub-menu component

I wanted it be smart/flexible when it comes adding items but unfortunately this doesn't work with the overflow scrolling, it needs a width for the whole sub-menu or for the items. You'll notice in the CSS I've created a few modifiers to declare how many items will be in the sub-menu which feels like a fairly innocous solution.

### CODE REVIEW

File: src/css/base.document.css [hunk 1]
```diff
@@ -4,6 +4,7 @@ html {
   line-height: var(--font-leading-none);
   -moz-osx-font-smoothing: grayscale;
   -webkit-font-smoothing: antialiased;
+  scroll-behavior: smooth;
```
[the remark this answers is not in the document]
Comment (adamduncan) [reply src/css/base.document.css:7]: We might consider putting [`scroll-margin-top`](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-margin-top) on the elements that we're smooth-scrolling to, to get a nice anchor depth that factors in the padding/sticky header

At the moment we're putting `id="performance"` on the section h2s. If we put them in the `<section>` instead, and applied a `scroll-margin-top: [header height]` to them, it'd line up a treat 👍

**Review by adamduncan — APPROVED** [review]
Looking slick, @Pete-Robelou. Merge away when you're ready.

File: src/css/base.document.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pete-Robelou) [reply src/css/base.document.css:7]: Yeah, good shout.

I've created a `--size-scroll-top` custom property.

File: src/css/component.sub-menu.css [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    overflow-x: auto;
+    padding: var(--size-0);
+  }
+
+  .sub-menu > li {
+    position: relative;
+  }
+
+  .sub-menu--2 > li {
+    flex: 1 0 var(--size-1-2);
```
[the remark this answers is not in the document]
Comment (Pete-Robelou) [reply src/css/component.sub-menu.css:30]: Quantity queries are working like a charm.

There's a postCSS plugin too https://github.com/pascalduez/postcss-quantity-queries. Maybe something to explore in the future.

**Pull request merged**